From bbc6938275a5204a8cfc49b6e80019235ffde5dc Mon Sep 17 00:00:00 2001
From: Peter Qiu <zqiu@chromium.org>
Date: Thu, 19 Feb 2015 15:10:17 -0800
Subject: [PATCH] CHROMIUMOS: Ignore driver's feature flag for low priority
 scan

Some drivers (such as brcmfmac and mwifiex) doesn't advertise its low
priority scan support (NL80211_FEATURES_LOW_PRIORITY_SCAN) through its
wiphy info, even though it is supported in the driver.

Trigger low priority scan regardless of the features that driver
advertised. Always assume driver supports low priority scan.
---
 src/drivers/driver_nl80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index b99d296..eb4b2a5 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -5134,7 +5134,7 @@ nl80211_scan_common(struct wpa_driver_nl80211_data *drv, u8 cmd,
 		scan_flags |= NL80211_SCAN_FLAG_FLUSH;
 	}
 
-	if (params->low_priority && drv->have_low_prio_scan) {
+	if (params->low_priority) {
 		wpa_printf(MSG_DEBUG,
 			   "nl80211: Add NL80211_SCAN_FLAG_LOW_PRIORITY");
 		scan_flags |= NL80211_SCAN_FLAG_LOW_PRIORITY;
-- 
2.2.0.rc0.207.ga3a616c

